core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceFactory.java

```diff
@@ -242,6 +242,12 @@ private static HttpStatus defaultResponseStatus(Method method) {
                                                 List<ResponseConverterFunction> baseResponseConverters,
                                                 List<ExceptionHandlerFunction> baseExceptionHandlers) {
 
+        if (!KotlinUtil.isArmeriaKotlinDependencyPresent() && KotlinUtil.maybeSuspendingFunction(method)) {
```

Comment:
How about just checking whether `CALL_KOTLIN_SUSPENDING_METHOD` is `null` instead of introducing another field?